cli/azd/internal/telemetry/storage_test.go

```diff
@@ -0,0 +1,247 @@
+package telemetry
+
+import (
+	"fmt"
+	"os"
+	"path/filepath"
+	"testing"
+	"time"
+
+	"github.com/azure/azure-dev/cli/azd/pkg/osutil"
+	"github.com/benbjohnson/clock"
+	"github.com/stretchr/testify/assert"
+	"golang.org/x/exp/maps"
+)
+
+// The tests in this file intentionally interacts with the filesystem (important implementation detail).
+// As such, it might be susceptible to filesystem related failures and also general slowness.
+
+const fileExtension = ".itm"
+
+type tempFolder struct {
+	name string
+}
+
+func newTempDir(t *testing.T, name string) tempFolder {
```

Comment:
https://pkg.go.dev/testing#T.TempDir does what you want, I think.

Reply:
Neat